Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1505,21 +1509,58 @@ function createAnimatedComponent(Component: any): any {
 
     componentWillUnmount() {
       this._propsAnimated && this._propsAnimated.__detach();
+      this._detachNativeEvents(this.props);
     }
 
     setNativeProps(props) {
       this._component.setNativeProps(props);
     }
 
     componentWillMount() {
-      this.attachProps(this.props);
+      this._attachProps(this.props);
     }
 
     componentDidMount() {
       this._propsAnimated.setNativeView(this._component);
+
+      this._attachNativeEvents(this.props);
+    }
+
+    _attachNativeEvents(newProps) {
+      if (newProps !== this.props) {
```

Comment:
I think the common case is to create a new event in every render so it should be fine to reattach everytime, I'll take another look see if I can improve it though.
``` js
<ScrollView
  onScroll={Animated.event({...})
/>
```